spec/std/crypto/subtle_spec.cr

```diff
@@ -17,11 +17,11 @@ describe "Subtle" do
 
   it "compares constant time bytes on equality" do
     data = [
-      {"a" => 0x00_u8, "b" => 0x00_u8, "result" => 1},
-      {"a" => 0x00_u8, "b" => 0x01_u8, "result" => 0},
-      {"a" => 0x01_u8, "b" => 0x00_u8, "result" => 0},
-      {"a" => 0xff_u8, "b" => 0xff_u8, "result" => 1},
-      {"a" => 0xff_u8, "b" => 0xfe_u8, "result" => 0},
+      {"a" => 0x00_u8, "b" => 0x00_u8, "result" => 1_u8},
+      {"a" => 0x00_u8, "b" => 0x01_u8, "result" => 0_u8},
+      {"a" => 0x01_u8, "b" => 0x00_u8, "result" => 0_u8},
+      {"a" => 0xff_u8, "b" => 0xff_u8, "result" => 1_u8},
+      {"a" => 0xff_u8, "b" => 0xfe_u8, "result" => 0_u8},
```

Comment:
Fixes the `UInt8 | Int32` union showing up on `constant_time_byte_eq` below

Reply:
Let's keep this PR focused on adding type restrictions. Fixing a spec should be a separate change. I have a small refactor already lined up for this.

Reply:
Sounds good, I'll remove the type restrictions on `constant_time_byte_eq` below then until that's in.